CHANGELOG.md

```diff
@@ -6,6 +6,93 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Unreleased
+- When querying the summary endpoint from Kubelet, if the default network
+  interface is not eth0 then summary endpoint for Kubelet doesn't return the
+  metrics as we expect them. We rely on them being a direct member of the
+  "network" object. See rxBytes, txBytes and rxErrors in the following example:
+
+```
+"network": {
+ "time": "2020-06-04T10:01:15Z",
+ "name": "eth0",
+ "rxBytes": 207909096,
+ "rxErrors": 0,
+ "txBytes": 8970981,
+ "txErrors": 0,
+ "interfaces": [
+  {
+   "name": "eth0",
+   "rxBytes": 207909096,
+   "rxErrors": 0,
+   "txBytes": 8970981,
+   "txErrors": 0
+  },
+  {
+   "name": "ip6tnl0",
+   "rxBytes": 0,
+   "rxErrors": 0,
+   "txBytes": 0,
+   "txErrors": 0
+  },
+  {
+   "name": "tunl0",
+   "rxBytes": 0,
+   "rxErrors": 0,
+   "txBytes": 0,
+   "txErrors": 0
+  }
+ ]
+}
+```
+
+  This scenario only happens when the default interface of the node is
+  eth0. Kubernetes source code has it hardcoded that eth0 is the default.
+  In the following example you can see that we only have network metrics inside
+  the interfaces list, in this case there is no eth0 on the Node and the
+  default interface is ens5:
+
+```
+"network": {
+ "time": "2020-06-04T10:01:15Z",
+ "name": "",
+ "interfaces": [
+  {
+   "name": "ens5",
+   "rxBytes": 207909096,
+   "rxErrors": 42,
+   "txBytes": 8970981,
+   "txErrors": 24
+  },
+  {
+   "name": "ip6tnl0",
+   "rxBytes": 0,
+   "rxErrors": 0,
+   "txBytes": 0,
+   "txErrors": 0
+  },
+  {
+   "name": "tunl0",
+   "rxBytes": 0,
+   "rxErrors": 0,
+   "txBytes": 0,
+   "txErrors": 0
+  }
+ ]
+```
+
+  In cases like this, the integration will look for the default interface
+  inside the interfaces list and use those values. The default interface name
+  is retrieved from the network route file (default /proc/net/route).
+
+  When running the unprivileged version of the integration we don't have access
+  to the route file, the integration won't be able to get the default interface
+  name and won't send network metrics for the Node unless the Node has a
+  network interface called eth0.
+
+  This issue affects the pods with hostNetwok, since they shared the network
+  interfaces with the Node. If there is no eth0 interface pods with hostNetwork
+  won't be sending network metrics, the fallback to the default interface won't
+  be applied in this case.
```

Comment:
well it would be the same interface we already have for the node, so we can easily send the same data for pods with hostNetwork as we are doing for the node, the thing is that it was decided to not send the metrics for the pods with hostNetwork.

Reply:
The point is that I think the `route` command can be ran without needing access to the file. 🤔